src/atlas/interpolation/method/binning/Binning.cc

```diff
@@ -130,7 +137,8 @@ std::vector<double> Binning::getAreaWeights(const FunctionSpace& fspace) const {
   // diagonal of 'area weights matrix', W
   std::vector<double> ds_aweights;
 
-  if (fspace.type().compare(functionspace::detail::NodeColumns::static_type()) == 0) {
+  if (grid_type_ == "CS-LFR") {
```

Comment:
Possibly this logic can instead be, without configuration of grid_type:
```c++
bool is_cubed_sphere {false};
if (auto nodes_fs = functionspace::NodeColumns(fspace)) {
  if (CubedSphereGrid(nodes_fs.mesh().grid()) {
    is_cubed_sphere = true;
  }
}
if (is_cubed_sphere) { 
  ...
```
Then please remove grid_type_ everywhere.

Reply:
Good point ... thanks.
... done.